### PR TITLE
at_cuongnguyen: Create page, feature and definitions for Services screen

### PR DESCRIPTION
**Pull request by at-cuongnguyen**

### CODE REVIEW

File: Training/src/test/java/at_cuongnguyen/Services.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    private MobileElement actionBar;
+
+    @AndroidFindBy(id = "menu_toggle_modes")
+    private MobileElement tvMenuToggleModes;
+
+    @AndroidFindBy(className = "android.widget.ListView")
+    private MobileElement listMenu;
+
+    //Find previous button of header bar
+    @iOSFindBy(xpath = "//*[@text='Daily Dozen' and @class='UIAButton' and ./parent::*[@text='Daily Dozen']]")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/Services.java:27]: just use xpath = "//*[@text='Daily Dozen

File: Training/src/test/java/at_cuongnguyen/Services.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ * @author at-cuong.nguyen
+ */
+public class Services extends BasePage {
+    @AndroidFindBy(id = "action_bar")
+    private MobileElement actionBar;
+
+    @AndroidFindBy(id = "menu_toggle_modes")
+    private MobileElement tvMenuToggleModes;
+
+    @AndroidFindBy(className = "android.widget.ListView")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/Services.java:23]: where IOS?

File: Training/src/test/java/at_cuongnguyen/Services.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+import org.openqa.selenium.By;
+
+/**
+ * for screen
+ *
+ * @author at-cuong.nguyen
+ */
+public class Services extends BasePage {
+    @AndroidFindBy(id = "action_bar")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/Services.java:17]: define on Android

File: Training/src/test/java/at_cuongnguyen/Services.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    private MobileElement BtnVideo;
+
+    @iOSFindBy(className = "UIAImage")
+    @AndroidFindBy(id = "food_info_image")
+    private MobileElement ImgFood;
+
+    @AndroidFindBy(id = "change_units_button")
+    private MobileElement BtnUnit;
+
+    @iOSFindBy(xpath = "//*[@text='Serving Sizes' and @class='UIAStaticText' and ./parent::*[@class='UIATable']]")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/Services.java:45]: just use xpath = "//*[@text='Serving Sizes'

File: Training/src/test/resources/features/at_cuongnguyen/Services.feature [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      When I click on Back button
+      Then I move to previous screen
+
+    Scenario: Check click on Videos button
+      When I click on Videos button
+      Then Open the url in new tab correctly with param-request
+
+    Scenario: Check [Service] area
+      When I move to [service] area
+      Then I check the [Service type] text is Services
```
Comment (at-anhquach) [inline Training/src/test/resources/features/at_cuongnguyen/Services.feature:20]: add enter line

File: Training/src/test/java/at_cuongnguyen/Services.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        }
+        return this;
+    }
+
+    public String getTextTitle() {
+        return actionBar.findElement(By.className("android.widget.TextView")).getText();
+    }
+
+    public Services clickMenuButton() {
+        actionBar.findElement(By.className("android.widget.ImageView")).click();
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/Services.java:80]: where IOS?

File: Training/src/test/java/at_cuongnguyen/Services.java [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    @Override
+    public BasePage open() {
+        if (!isPageDisplayed()) {
+            getDriver().launchApp();
+        }
+        return this;
+    }
+
+    public String getTextTitle() {
+        return actionBar.findElement(By.className("android.widget.TextView")).getText();
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/Services.java:76]: where IOS?

File: Training/src/test/java/at_cuongnguyen/Android_Services.java [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+ * for screen
+ *
+ * @author at-cuong.nguyen
+ */
+public class Android_Services extends BasePage {
+    @AndroidFindBy(id = "action_bar")
+    @iOSFindBy(className = "UINavigationBar")
+    public MobileElement actionBar;
+
+    @AndroidFindBy(className = "UIATable")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/Android_Services.java:20]: incorrect className

File: Training/src/test/java/at_cuongnguyen/Android_Services.java [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    private MobileElement container;
+
+    //@AndroidFindBy(id = "menu_toggle_modes")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/Android_Services.java:23]: delete commnets code

File: Training/src/test/java/at_cuongnguyen/Android_Services.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+    @iOSFindBy(xpath = "//*[@text='VIDEOS'")
+    public MobileElement btnVideo;
+
+    @iOSFindBy(className = "UIAImage")
+    @AndroidFindBy(id = "food_info_image")
+    public MobileElement imgFood;
+
+    @AndroidFindBy(id = "change_units_button")
+    @iOSFindBy(id = "IMPERIAL")
+    @iOSFindBy(id = "METRIC")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/Android_Services.java:51]: why use 2 iOSFindBy?

File: Training/src/test/java/at_cuongnguyen/Android_Services.java [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+import io.appium.java_client.pagefactory.AndroidFindBy;
+import io.appium.java_client.pagefactory.iOSFindBy;
+import org.openqa.selenium.By;
+
+/**
+ * for screen
+ *
+ * @author at-cuong.nguyen
+ */
+public class Android_Services extends BasePage {
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/Android_Services.java:15]: wrong name file Android_Services -> ServicesPage

File: Training/src/test/java/at_cuongnguyen/iOS_Services.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+    public iOS_Services clickUnitButton(){
+        btnUnit.click();
+        return this;
+    }
+
+    public iOS_Services clickPreviousButton() {
+        actionBar.findElement(By.id("Daily Dozen")).click();
+        return this;
+    }
+
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/iOS_Services.java:55]: delete one line

File: Training/src/test/java/at_cuongnguyen/Services.java (hunk 1 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/Services.java]: fixed

File: Training/src/test/java/at_cuongnguyen/Services.java (hunk 2 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/Services.java]: fixed

File: Training/src/test/java/at_cuongnguyen/Services.java (hunk 3 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/Services.java]: fixed

File: Training/src/test/java/at_cuongnguyen/Services.java (hunk 4 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/Services.java]: fixed

File: Training/src/test/java/at_cuongnguyen/Services.java (hunk 6 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/Services.java]: fixed

File: Training/src/test/java/at_cuongnguyen/Services.java (hunk 7 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/Services.java]: fixed

File: Training/src/test/resources/features/at_cuongnguyen/Services.feature (hunk 5 above)
Comment (at-cuongnguyen) [reply Training/src/test/resources/features/at_cuongnguyen/Services.feature:20]: FIXED

File: Training/src/test/java/at_cuongnguyen/Android_Services.java (hunk 8 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/Android_Services.java]: fixed

File: Training/src/test/java/at_cuongnguyen/Android_Services.java (hunk 9 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/Android_Services.java]: fixed

File: Training/src/test/java/at_cuongnguyen/Android_Services.java (hunk 10 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/Android_Services.java]: fixed

File: Training/src/test/java/at_cuongnguyen/Android_Services.java (hunk 11 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/Android_Services.java]: fixed. Add more page [Service] , [AndroidServicePage], [iOSServicepage]

File: Training/src/test/java/at_cuongnguyen/iOS_Services.java (hunk 12 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/iOS_Services.java]: Fixed

**Review by at-cuongnguyen — COMMENTED** [review]
Fixed all comment from @at-anhquach

File: Core/src/main/java/at/core/AppiumController.java [hunk 13]
```diff
@@ -109,8 +109,8 @@ private synchronized void startDefaultServer() throws MalformedURLException {
     private Map<String, String> defaultAndroidParameters() {
         Map<String, String> parameters = new HashMap<>();
         parameters.put(MobileCapabilityType.PLATFORM_NAME, "android");
-        parameters.put(MobileCapabilityType.DEVICE_NAME, "Pixel XL API 27");
-        parameters.put(MobileCapabilityType.PLATFORM_VERSION, "8.1");
+        parameters.put(MobileCapabilityType.DEVICE_NAME, "Pixel API 28 2");
+        parameters.put(MobileCapabilityType.PLATFORM_VERSION, "9.0");
```
Comment (at-anhquach) [inline Core/src/main/java/at/core/AppiumController.java:113]: Don't change this file

File: Training/src/test/java/ServicesDefinition.java [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+
+        When("^I check the header bar on services screen$", () -> Assert.assertTrue(true));
+
+        Then ("^ Having Videos button which on the right of header bar$", () -> Assert.assertTrue(service.isbtnvideoContain()));
+
+       // When("^I Click on [Previous button]", () -> service.
+
+
+    }
+
```
Comment (at-anhquach) [inline Training/src/test/java/ServicesDefinition.java:37]: delete this line

File: Training/src/test/java/at_cuongnguyen/ServicesPage.java [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+    public ServicesPage(MobileDriver driver) {
+        super(driver);
+    }
+
+    @AndroidFindBy(id = "action_bar")
+    @iOSFindBy(className = "UIANavigationBar")
+    public MobileElement actionBar;
+
+    //This element just only iOS containing
+    @iOSFindBy(className ="UIAStaticText")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/ServicesPage.java:26]: define on IOS page and use another way to find element, UIAStaticText duplicate

File: Training/src/test/java/at_cuongnguyen/ServicesPage.java [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+
+    @AndroidFindBy(id = "food_info_videos")
+    @iOSFindBy(xpath = "//*[@text='VIDEOS'")
+    public MobileElement btnVideo;
+
+    @iOSFindBy(className = "UIAImage")
+    @AndroidFindBy(id = "food_info_image")
+    public MobileElement imgFood;
+
+    @AndroidFindBy(id = "change_units_button")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/ServicesPage.java:45]: define on Android Page

File: Training/src/test/java/at_cuongnguyen/ServicesPage.java [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+    public MobileElement btnUnit;
+
+    @iOSFindBy(id = "IMPERIAL")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/ServicesPage.java:48]: define on IOS Page

File: Training/src/test/java/at_cuongnguyen/ServicesPage.java [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+    public MobileElement btnImperial;
+
+    @iOSFindBy(id = "METRIC")
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/ServicesPage.java:51]: define on IOS Page

File: Training/src/test/java/at_cuongnguyen/ServicesPageAndroid.java [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+        return this;
+    }
+
+    //public String getTextTvMenuToggleModes() {return tvMenuToggleModes.getText(); }
+
+    //public boolean isListMenuDisplay() {return isForElementPresent(listMenu);}
+
+    //public String getTextOfNameFood(int pos) {return container.getText();}
+
+    //public String clickIcCalendar(int pos) {return container.findElements(By.className("UIAStaticText")).get(pos).getText();}
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/ServicesPageAndroid.java:71]: delete comment code

File: Training/src/test/java/at_cuongnguyen/ServicesPageIOS.java [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+        if (btnImperial.isDisplayed()) {
+            btnImperial.click();
+        }
+        else  btnMetric.click();
+    }
+    public ServicesPageIOS clickPreviousButton() {
+        actionBar.findElement(By.id("Daily Dozen")).click();
+        return this;
+    }
+}
```
Comment (at-anhquach) [inline Training/src/test/java/at_cuongnguyen/ServicesPageIOS.java:59]: enter line

File: Training/src/test/java/page/exam/HomePage.java [hunk 21]
```diff
@@ -42,6 +42,7 @@ public String getTextTitle() {
         return actionBar.findElement(By.className("android.widget.TextView")).getText();
     }
 
+
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/HomePage.java:45]: delete line

File: Core/src/main/java/at/core/AppiumController.java (hunk 13 above)
Comment (at-cuongnguyen) [reply Core/src/main/java/at/core/AppiumController.java:113]: Done

File: Training/src/test/java/ServicesDefinition.java (hunk 14 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/ServicesDefinition.java:37]: Done

File: Training/src/test/java/at_cuongnguyen/ServicesPage.java (hunk 17 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/ServicesPage.java:48]: Done

File: Training/src/test/java/at_cuongnguyen/ServicesPage.java (hunk 16 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/ServicesPage.java:45]: Done

File: Training/src/test/java/at_cuongnguyen/ServicesPage.java (hunk 18 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/ServicesPage.java:51]: Done

File: Training/src/test/java/at_cuongnguyen/ServicesPageAndroid.java (hunk 19 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/ServicesPageAndroid.java:71]: done

File: Training/src/test/java/at_cuongnguyen/ServicesPageIOS.java (hunk 20 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/at_cuongnguyen/ServicesPageIOS.java:59]: done

File: Training/src/test/java/page/exam/HomePage.java (hunk 21 above)
Comment (at-cuongnguyen) [reply Training/src/test/java/page/exam/HomePage.java:45]: done

File: Core/src/main/java/at/core/AppiumController.java (hunk 13 above)
Comment (at-cuongnguyen) [reply Core/src/main/java/at/core/AppiumController.java:113]: This AppiumController.java I pushed it wrong, So I won't push it .

**Pull request merged**